### ISSUE TITLE
Excessive writes for response headers

### ISSUE THREAD

**Issue by xvjau** [BoostGSoC14/boost.http#54]
Running a unit-test, I noticed that for every header, the header name and the header value as well as the \r\n at the end would result in three separate writes.  This means three Kernel-Space/User-Space switches for each single header.

**Comment by vinipsmaker** [comment]
I send multiple buffers: https://github.com/BoostGSoC14/boost.http/blob/26a2c2c9ea8dd362623a0a7fd7c4c1490fb5dd90/include/boost/http/socket.ipp#L304

This must be the cause. I didn't know `asio_handler_allocate` at the time I implemented it. On the later client-side support, I've used `asio_handler_allocate` and allocated a single buffer to the whole output message buffer: https://github.com/BoostGSoC14/boost.http/blob/26a2c2c9ea8dd362623a0a7fd7c4c1490fb5dd90/include/boost/http/socket.ipp#L591

I plan to refactor output response message buffer to also use this strategy.

It's easy, so it shouldn't take more than half-a-day to implement it, but you've issued quite a few tasks for me, so I need to schedule which ones will be implemented first.
